### PR TITLE
test if inc_count is a number

### PR DESCRIPTION
**Pull request by Nat2llaw**
add one more test to ncnews 6, discussed with Michael already

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -43,57 +43,70 @@ describe("check for correct pathing", () => {
...
-    describe("PATCH/api/articles/:article_id", () => {
-      test("200: return article by id with increased vote value", () => {
+      test("200: return article with comment_count", () => {
         return request(app)
-          .patch("/api/articles/1")
-          .send({ inc_votes: 10 })
+          .get("/api/articles/1")
           .expect(200)
-          .then(({ body: { article } }) => {
+          .then(({ body: [ article ] }) => {
```
Comment (corkesw) [inline __tests__/app.test.js:52]: if we are only returning one article then I would not expect it to be in an array

File: db/model/model.js [hunk 2]
```diff
@@ -1,33 +1,44 @@
...
-            return Promise.reject({status: 400, msg: 'Id not found'})
-        }
-        return rows
-    })
-}
-
-exports.fetchUsers = () => {
   return db
-    .query(`SELECT * FROM users`)
+    .query(`ALTER TABLE articles ADD comment_count INT NOT NULL DEFAULT 0 `)
```
Comment (corkesw) [inline db/model/model.js:11]: We don't want to alter the table here, we can use a `JOIN` to get a comment count from the combined articles and comments table. There is an example here [SQL - Foreign keys, Joins and Aggregate Functions](https://notes.northcoders.com/courses/js-back-end/sql-relationships)

File: db/model/model.js [hunk 3]
```diff
@@ -1,33 +1,44 @@
...
+        );
+    })
+    .then(() => {
+      return db.query(`SELECT * FROM articles WHERE article_id=$1`, [id]);
+    })
     .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 400, msg: "Id not found" });
+      }
       return rows;
```
Comment (corkesw) [inline db/model/model.js:25]: We're only returning one thing from this endpoint so remove it from the array first!

File: __tests__/app.test.js (hunk 1 above)
Comment (Nat2llaw) [reply __tests__/app.test.js:52]: only works using [ ]. Tried early with Michael and it was the only way to get a singular result

**Comment by Nat2llaw** [comment]
added the left join to get a singular result with the comment count

**Review by corkesw — COMMENTED** [review]
This solution for the comment_count is much better! We are still returning the rows from the model though and, as we are only sending back a single object it doesn't make sense to return an array. You should change the test to expect a single article object on the key of article and amend you model to achieve this. Once you have actioned this last comment you can merge when ready - you do not need to submit another PR.

**Pull request merged**